pallets/pallet-bonded-coins/src/curves/mod.rs

```diff
@@ -111,6 +111,31 @@ pub(crate) fn convert_to_fixed<T: Config>(
 	let decimals = 10u128
 		.checked_pow(u32::from(denomination))
 		.ok_or(ArithmeticError::Overflow)?;
-	let scaled_x = x.checked_div(decimals).ok_or(ArithmeticError::DivisionByZero)?;
```

Comment:
This was dropping everything that is not a multiple of the denomination - that can't be the point of conversion to fixed, right @Ad96el ?

Reply:
Yes, this is a bug. Initially, I have first converted X into a fixed and then divided it by the denomination.